home-work-function-pro.js

```diff
@@ -0,0 +1,211 @@
+console.log(addThemAll(1, 2, 3, 4)); // 10
+console.log(addThemAll(5, 5, 10)); // 20
+
+
+// 2. Задача на використання замикання
+
+const multiply = a => b => a * b;
+// або:
+
+const multiply = a => (b) {
+  return a * b;
+};
+
+console.log(multiply(5)(5)); // Output: 25
+console.log(multiply(2)(-2)); // Output: -4
+console.log(multiply(4)(3)); // Output: 12
+
+// 3. Напишіть функцію яка буде використовуватись для сортування масиву фільмів
+const movies = [
+  {
+  movieName: 'The Thing',
+  releaseYear: 1982,
+  directedBy: 'Carpenter',
+  runningTimeInMinutes: 109,
+  },
+  {
+  movieName: 'Aliens',
+  releaseYear: 1986,
+  directedBy: 'Cameron',
+  runningTimeInMinutes: 137,
+  },
+  {
+  movieName: 'Men in Black',
+  releaseYear: 1997,
+  directedBy: 'Sonnenfeld',
+  runningTimeInMinutes: 98,
+  },
+  {
+  movieName: 'Predator',
+  releaseYear: 1987,
+  directedBy: 'McTiernan',
+  runningTimeInMinutes: 107,
+  },
+  ];
+  
+const byProperty = (property, direction) => (a, b) => {
+  const valueA = a[property];
+  const valueB = b[property];
+  let comparison = 0;
+
+  if (valueA > valueB) {
+    comparison = 1;
+  }
+  if (valueA < valueB) {
+    comparison = -1;
+  }
+  if (direction === '<') {
+    comparison = comparison * -1;
+  }
+  return comparison;
+};
+
+  
+  
+  console.log(movies.sort(byProperty('releaseYear', '>'))); //виведе масив фільмів посортованих по року випуску, від старішого до новішого*
+  console.log(movies.sort(byProperty('runningTimeInMinutes', '<')));  //виведе масив фільмів посортованих по їх тривалості, від найдовшого до найкоротшого*
+  console.log(movies.sort(byProperty('movieName', '>'))); //виведе масив фільмів посортованих по назві, в алфавітному порядку*
+  
+  // 4 Напишіть функцію detonatorTimer(delay), яка виводить в консоль число кожну секунду, починаючи з delay (ціле число) і в кінці замість 0 виведе 'BOOM!'
+
+  // Використовуючи setInterval
+
+  function detonatorTimer(delay) {
+    let remainingSeconds = delay;
+    const timerId = setInterval(() => {
+      let output = remainingSeconds
+      if (remainingSeconds > 0) {
+        remainingSeconds--;
+      } else { 
+      if (remainingSeconds == 0) {
+        output = 'Boom';
+        clearInterval(timerId);
+      }}
+      console.log(output)
+    }, 1000);
+  }
+  
+  detonatorTimer(3);
+
+  // або
+
+  function detonatorTimer(delay) {
+    let remainingSeconds = delay;
+    const timerId = setInterval(() => {
+      let output = remainingSeconds > 0 ? remainingSeconds : 'BOOM!';
+      console.log(output);
+      remainingSeconds--;
+      if (remainingSeconds < 0) {
+        clearInterval(timerId);
+      }
+    }, 1000);
+  }
+  
+  detonatorTimer(3);
+
+  // Використовуючи вкладений setTimeout
+
+  function detonatorTimer(delay) {
+    let output = delay;
+  if (delay > 0) {
+    setTimeout(function() {
+      detonatorTimer(delay - 1);
+    }, 1000);
+  } else {
+    output = 'Boom!';
+  }
+  console.log(output);
```

Comment:
strange indentation

Reply:
In these [commit](https://github.com/ilyaivanish/ilya-ivanish-prjctr/pull/2/files/aebfca44727a3241dc16c18f06bc53d01ed2b0d0#r1147344740) you ask me to use only once console.log. So I had to move it here